tests/e2e/Services/Projects/ProjectsCustomServerTest.php

```diff
@@ -54,5 +53,21 @@ public function testCreateProjectRule()
         ]);
 
         $this->assertEquals(400, $response['headers']['status-code']);
+
+        // prevent functions domain
+        $response = $this->client->call(Client::METHOD_POST, '/proxy/rules', $headers, [
+            'resourceType' => 'function',
+            'domain' => $functionsDomain,
+        ]);
+
+        $this->assertEquals(400, $response['headers']['status-code']);
+
+        // prevent sites domain
+        $response = $this->client->call(Client::METHOD_POST, '/proxy/rules', $headers, [
+            'resourceType' => 'site',
+            'domain' => $sitesDomain,
+        ]);
+
+        $this->assertEquals(400, $response['headers']['status-code']);
```

Comment:
Let's add a few more tests. We can run all reservedDomains in a loop. pseudo code:

```
denyList = [localhost, appwrite, $sitesDomain, $functionsDomain, $domain] // $domain is from _APP_DOMAIN env var

for denyList as domain {
  response = call(post proxy/roles, domain=domain)
  assertEquals(400, response.code)
}
```